wicket-jquery-ui-parent/wicket-jquery-ui-calendar/src/main/java/com/googlecode/wicket/jquery/ui/calendar6/Calendar.java

```diff
@@ -0,0 +1,270 @@
+	 * @param model the {@link CalendarModel}
+	 */
+	public Calendar(String id, CalendarModel model)
+	{
+		this(id, model, new Options());
+	}
+
+	/**
+	 * Constructor
+	 *
+	 * @param id the markup id
+	 * @param model the {@link CalendarModel}
+	 * @param options the {@link Options}. Note that 'selectable' and 'selectHelper' options are set by overriding {@link #isSelectable()} (default is false)
+	 */
+	public Calendar(String id, CalendarModel model, Options options)
+	{
+		super(id, model);
+
+		this.options = Args.notNull(options, "options");
+	}
+
+	/**
+	 * Gets the calendar's model
+	 *
+	 * @return a {@link CalendarModel}
+	 */
+	public CalendarModel getModel()
+	{
+		return (CalendarModel) this.getDefaultModel();
+	}
+
+	// Methods //
+
+	/**
+	 * Adds a {@link EventSource}, can be a {@link GoogleCalendar}
+	 *
+	 * @param source the {@link EventSource},
+	 */
+	public void addSource(EventSource source)
+	{
+		if (this.sources == null)
+		{
+			this.sources = Generics.newArrayList();
+		}
+
+		this.sources.add(source);
+	}
+
+	/**
+	 * Re-fetches and refreshes the events currently available in the selected view.
+	 *
+	 * @param handler the {@link IPartialPageRequestHandler}
+	 */
+	public void refresh(IPartialPageRequestHandler handler)
+	{
+		handler.appendJavaScript(String.format("document.querySelector('%s').calendar.refetchEvents();", JQueryWidget.getSelector(this)));
+	}
+
+	// Properties //
+
+	@Override
+	public boolean isSelectable()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isDateClickEnabled()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isEventClickEnabled()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isEventDropEnabled()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isEventResizeEnabled()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isObjectDropEnabled()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isViewDidMountEnabled()
+	{
+		return false;
+	}
+
+	@Override
+	public CharSequence getEventDropPrecondition()
+	{
+		return "";
+	}
+
+	@Override
+	public CharSequence getEventResizePrecondition()
+	{
+		return "";
+	}
+
+	// Events //
+
+	@Override
+	protected void onInitialize()
+	{
+		super.onInitialize();
+
+		this.modelBehavior = this.newCalendarModelBehavior(this.getModel());
+		this.behavior = new CalendarBehavior(JQueryWidget.getSelector(this), this.options, this);
+		this.add(this.behavior, this.modelBehavior);
+	}
+
+	@Override
+	public void onConfigure()
+	{
+		super.onConfigure();
+
+		// builds sources //
+		StringBuilder sourceBuilder = new StringBuilder();
+		sourceBuilder.append(String.format("{ url: '%s' }", Calendar.this.modelBehavior.getCallbackUrl()));
+
+		if (Calendar.this.sources != null)
+		{
+			for (EventSource source : Calendar.this.sources)
+			{
+				sourceBuilder.append(", ");
+				sourceBuilder.append(source.toString());
+			}
+		}
+
+		behavior.setOption("eventSources", String.format("[%s]", sourceBuilder.toString()));

```

Comment:
```suggestion
		behavior.setOption("eventSources", "[" + sourceBuilder.toString() + "]");

```
Shorter code and faster at runtime

Reply:
fixed